### ISSUE TITLE
Difficult to setup

### ISSUE THREAD

**Issue by alikaylan** [greghare/nook-dashboard#1]
Love the idea, and appreciate the implementation. Having no calendar, no need to do list, different weather app in HA (forecast home) makes changing the server.py modifications rather difficult. Would there be a simple version of the server.py file that just displays internal sensors switches, etc... Perhaps the original version was closer to that, but I don't see it there anymore... Thanks.

**Comment by greghare** [comment]
I had considered making it more universal in the future, so that everything can be configured from the config file, but for now just wanted to release what I had. 

If I understand correctly, you'd like to see just the current weather, forecast, and the table of entities with their states. Could you provide a link to the weather integration you are using? I may try to test with that and see if I can provide something like what you're looking for.

**Comment by alikaylan** [comment]
The interesting parts of my Lovelace, including the boiler plate weather card that came as default are below. The boiler plate weather card is the one [here](https://smarthomescene.com/blog/top-10-home-assistant-weather-cards/) as the first choice. Thank you so much.. 

      - type: light
        entity: light.second_bedroom_dimmer_tasmota
        name: Second Bedroom Light
        icon: mdi:ceiling-light
      - type: weather-forecast
        entity: weather.forecast_home
        show_forecast: true
      - type: vertical-stack
        cards:
          - type: entities
            title: Sunrise and Sunset
            show_header_toggle: false
            entities:
              - sensor.nextsunrise
              - sensor.nextsunset
      - type: entities
        entities:
          - entity: sensor.ble_temperature_govee
          - entity: sensor.ble_humidity_govee
          - entity: sensor.ble_battery_govee
        title: Outside Thermometer
      - type: entities
        entities:
          - entity: sensor.ble_temperature_govee_inside
          - entity: sensor.ble_humidity_govee_inside
          - entity: sensor.ble_battery_govee_inside
        title: Inside Thermometer

      - type: entities
        entities:
          - sensor.bh1750_illuminance
        title: Living Room Illumination
        state_color: true

      - type: vertical-stack
        cards:
          - type: entities
            title: HVAC Economy Setback
            show_header_toggle: false
            entities:
              - entity: switch.living_room_hvac_econ_setback
                name: Living Room
                icon: mdi:sofa
              - entity: switch.kitchen_hvac_econ_setback
                name: Kitchen
                icon: mdi:countertop
              - entity: switch.bedroom_hvac_econ_setback
                name: Bedroom
                icon: mdi:bed-king

      - type: vertical-stack
        cards:
          - type: entities
            title: Sunrise and Sunset
            show_header_toggle: false
            entities:
              - sensor.nextsunrise
              - sensor.nextsunset

**Comment by greghare** [comment]
I'm working on updating the config file to include many more options, including a list of the entities to display, configurable weather and tide entities, and I'm planning to add options to disable the todo, calendar, and tide for those that want a full width display of the weather and device state table, but have no need for other information.

I appreciate the feedback to help make this more useful!

**Comment by greghare** [comment]
Here's what the configuration file I'm working with currently looks like (I have not pushed the updated code yet)

![image](https://user-images.githubusercontent.com/6226804/234163310-82e0e509-6fdc-4a1c-8399-101fb9690650.png)

**Comment by jeakob** [comment]
Hey @greghare I was wondering if you had an update on the new code that you mentioned?

**Comment by kentkravitz** [comment]
Let's get the code up to date, then get this working as a home assistant addon.

**Comment by Hadeone-ha** [comment]
Hi, any update? I am also stuck trying to set up because I can't use openweather.  Thanks

**Comment by greghare** [comment]
Hey @Hadeone-ha, yesthanks for reaching out! I'm going to try to spend some time later today getting it back to a good working state, and making it a bit easier to setup. I'll let you know once I get it fixed and have some new instructions written.

But to answer your question, it's likely because it's trying to use openweathermap. You could try editing this line and change "/states/weather.openweathermap" to "/states/weather.NAME_OF_YOUR_WEATHER_ENTITY", but if it doesn't have the same fields as openweathermap it probably won't work

        response = requests.get(HA_API + "/states/weather.openweathermap", headers=headers, verify=False)

**Comment by andiohn** [comment]
Awesome! I'll be testing soon!

**Comment by Hadeone-ha** [comment]
@greghare thanks for coming back to this after so many years and keeping the project going.

**Comment by greghare** [comment]
Absolutely! I moved during this time and hadn't setup my dashboard again, so I wanted to get it back working too :)

**Comment by Hadeone-ha** [comment]
> Absolutely! I moved during this time and hadn't setup my dashboard again, so I wanted to get it back working too :)

@greghare I got it working with the new code however I have a very noob question I can't figure out.  How do you get the layout horizontal on ElectricSign?  I dont see an option and mine shows the dashboard vertically, all squished together. 

Thanks again!

**Comment by greghare** [comment]
@Hadeone-ha, not a noob question! It's actually harder than it should be, but I ended up installing an app called "Screen Rotator". I can't seem to find the APK download link I used, but I think I found it on either some APK site or xdaforums somewhere.

**Comment by Hadeone-ha** [comment]
In case anyone else needs more help on getting the nook set up initially, rooting, rotating the screen, here are my experiences. I have a Nook Simple Touch BNRV300 and a Glowlite BNRV350.  On the 300 I used NookManager 0.5.0 to root it.  This was very simple, ElectricSign was preinstalled after rooting, and in the settings for ReLaunch (the launcher that is installed after rooting) you can find an option to rotate your view to landscape.  This was all I needed to do get @greghare ’s dashboard working as intended.   On the 350 NookManager doesn’t work so I used the Phoenix project to root and install custom software (based on 1.1.5 fw).   I installed ElectricSign but wasn’t yet able to find a screen rotator app that works with Android 2.1.  Hope this helps.  Here is another useful link with instructions and screenshots of the rooting process https://ben-park.co.uk/2013/09/08/rooting-a-nook-my-experiences/

My next issue is that these nooks both have dead batteries that have expanded and bulged the rear case.  I’m trying to figure out how to get the nook to run without a battery.  It doesnt turn on when plugged in wo a battery or even with the chip from the battery plugged in after I removed it from the bad battery pouch.  Info here https://www.instructables.com/HOW-TO-REPLACE-a-NOOK-SIMPLE-TOUCH-NST-BATTERY-WIT/

This isnt a big issue because replacement batteries are available but I plan to mount with permanent power hooked up.

**Comment by greghare** [comment]
I will see if I can get some time to hook mine up with ADB and maybe I can pull the screen rotator APK for the one I'm using.

That's too bad about the batteries. And unfortunately leaving batteries plugged in just deteriorates them overtime. If you're good with electronics and soldering you might be able to rig up a 3.7v power supply directly to the board.

**Comment by Hadeone-ha** [comment]
That’s a good idea. 3.7v connected to the battery “chip” should work.  Not sure how it will behave when usb is plugged in as well, if needed to transfer files or anything but I’ll give it s shot and report back .  Thanks
